utils/utils.c

```diff
@@ -886,6 +886,23 @@ char *absolute_dirname(const char *path, char *resolved_path)
 	return resolved_path;
 }
 
+char *uftrace_strerror_r(int errnum, char *buf, size_t buflen)
+{
+	long strerror_r_result = (long) strerror_r(errnum, buf, buflen);
+
+	if (strerror_r_result == 0)
+		/* XSI-compliant strerror_r succeed */
+		return buf;
+	else if (strerror_r_result > 0 && strerror_r_result < 4096) {
```

Comment:
This can be simply `else if (strerror_r_result < 4096)` as userspace pointer never goes to negative.